sky/engine/core/painting/RRect.dart

```diff
@@ -1,32 +1,234 @@
 part of dart_ui;
 
-// A rounded rectangle.
+// A rounded rectangle with the same radii for all four corners.
 class RRect {
-  RRect();
+  RRect._();
 
-  /// Initialize with the same radii for all four corners.
-  RRect.fromRectXY(Rect rect, double xRadius, double yRadius) {
+  /// Construct a rounded rectangle from its left, top, right, and bottom edges,
+  /// and the radii along its horizontal axis and its vertical axis.
+  RRect.fromLTRBXY(double left, double top, double right, double bottom, double radiusX, double radiusY) {
+    _value
+      ..[0] = left
+      ..[1] = top
+      ..[2] = right
+      ..[3] = bottom
+      ..[4] = radiusX
+      ..[5] = radiusY;
+  }
+
+  /// Construct a rounded rectangle from its bounding box and the radii along
+  /// its horizontal axis and its vertical axis.
+  RRect.fromRectXY(Rect rect, double radiusX, double radiusY) {
     _value
       ..[0] = rect.left
       ..[1] = rect.top
       ..[2] = rect.right
       ..[3] = rect.bottom
-      ..[4] = xRadius
-      ..[5] = yRadius;
+      ..[4] = radiusX
+      ..[5] = radiusY;
   }
 
   final Float32List _value = new Float32List(6);
 
+  /// The offset of the left edge of this rectangle from the x axis.
+  double get left => _value[0];
+
+  /// The offset of the top edge of this rectangle from the y axis.
+  double get top => _value[1];
+
+  /// The offset of the right edge of this rectangle from the x axis.
+  double get right => _value[2];
+
+  /// The offset of the bottom edge of this rectangle from the y axis.
+  double get bottom => _value[3];
+
+  /// The horizontal semi-axis of the corners.
+  double get radiusX => _value[4];
+
+  /// The vertical semi-axis of the corners.
+  double get radiusY => _value[5];
```

Comment:
In theory, you can have a separate radius for top and bottom as well.  That's pretty rare though (and we haven't supported it up to this point).